client/src/main.rs

```diff
@@ -582,7 +582,20 @@ fn fetch(
     );
     let mut store = DataStore::open_or_create(&opts.data_dir, interface)?;
     let api = Api::new(&config.server);
-    let State { peers, cidrs } = api.http("GET", "/user/state")?;
+    let (State { peers, cidrs }, server_is_reachable) = match api.http("GET", "/user/state") {
+        Ok(state) => (state, true),
+        Err(ureq::Error::Transport(_)) => {
+            log::warn!(
+                "Could not connect to the innernet server, proceeding with cached state instead."
+            );
+            let state = State {
+                peers: store.peers().to_vec(),
+                cidrs: store.cidrs().to_vec(),
+            };
```

Comment:
What happens if we don't have any cache here? I guess that will still bring the interface up, just report 0 peers? Do we want to explicitly bail instead?

Reply:
This will be the new behavior.

```
marek@Mac innernet % cargo build && sudo ./target/debug/innernet up tonari  
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.06s
[*] fetching state for tonari from server...

[E] Could not connect to the innernet server and there are no cached peers, bailing.
```

The interface itself is brought up first and keep up even if peer initialization fails and I guess that's intentional so that the subsequent fetches (which don't don't support bringing interface up automatically, for whatever reason) have a chance to succeed?

Reply:
New behavior looks good!

> The interface itself is brought up first and kept up even if peer initialization fails

Not sure if that was the _original_ intention, but now such a behavior is beneficial and depended on: you can start a program that binds to the listen port etc. So let's not change it here.